test/integration/physics_system.cc

```diff
@@ -1375,12 +1375,53 @@ TEST_F(PhysicsSystemFixture, NestedModelIndividualCanonicalLinks)
   EXPECT_DOUBLE_EQ(model12WorldPose.Z(), model10It->second.Z());
 }
 
+/////////////////////////////////////////////////
+TEST_F(PhysicsSystemFixture, DefaultPhysicsOptions)
+{
+  ignition::gazebo::ServerConfig serverConfig;
+
+  bool checked{false};
+
+  // Create a system to check components
+  test::Relay testSystem;
+  testSystem.OnPostUpdate(
+    [&checked](const gazebo::UpdateInfo &,
+    const gazebo::EntityComponentManager &_ecm)
+    {
+      _ecm.Each<components::World, components::PhysicsCollisionDetector,
+                components::PhysicsSolver>(
+        [&](const ignition::gazebo::Entity &, const components::World *,
+            const components::PhysicsCollisionDetector *_collisionDetector,
+            const components::PhysicsSolver *_solver)->bool
+        {
+          EXPECT_NE(nullptr, _collisionDetector);
+          if (_collisionDetector)
+          {
+            EXPECT_EQ("ode", _collisionDetector->Data());
```

Comment:
The default for `<collision_detector>` listed in http://sdformat.org/spec?ver=1.8&elem=physics#physics_dart is `fcl`. Is the SDF spec out of date?

Reply:
Yeah that's the default on Gazebo classic, but on Ignition, we started using Bullet (https://github.com/ignitionrobotics/ign-physics/commit/f298fb42eb000a89872bb1bc71b65cfd5e4e0179), and then ODE (https://github.com/ignitionrobotics/ign-physics/commit/8b00f01d4e3afe9af0f740e4f429ef3cb877b120).

Ultimately, I think the SDF spec shouldn't be opinionated with this kind of element, because each simulator may choose a different default value. This is slightly related to https://github.com/osrf/sdformat/issues/508.

Reply:
Ah, I see - thanks for sharing those links. I just took a look at the [Gazebo classic dependencies](http://gazebosim.org/tutorials?tut=install_dependencies_from_source&cat=install#SDFormat), and it looks like Gazebo 11 supports SDF specification 1.7, so should we at least update the SDF website starting at specification 1.8? I also wonder if for the time being (i.e., until https://github.com/osrf/sdformat/issues/508 is addressed), we should document somewhere about the different physics engine default parameters between Gazebo classic and ign-gazebo. I can also leave a comment on https://github.com/osrf/sdformat/issues/508 if it would be helpful to have these thoughts tracked there.

Reply:
> it looks like Gazebo 11 supports SDF specification 1.7, so should we at least update the SDF website starting at specification 1.8? 

I'm not sure this solves the problem. Ignition Citadel uses 1.7 and uses a different default from Gazebo classic. In the spirit of being simulator-agnostic, I think the SDF spec shouldn't dictate the default values for these options.

> we should document somewhere about the different physics engine default parameters between Gazebo classic and ign-gazebo.

Documented the options supported by Ignition on 16b1af4. And added the high-level comparison to Gazebo classic on https://github.com/ignitionrobotics/docs/pull/186.